src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,653 @@
+    /// pack venueLastRecordedProfit, venueTotalShares and profitToVconRatio
+    /// into a single slot for gas optimization
+
+    /// @notice last recorded profit index per venue
+    mapping(address => uint128) public venueLastRecordedProfit;
+
+    /// @notice last recorded VCON share price index per venue
+    mapping(address => uint128) public venueLastRecordedVconSharePrice;
+
+    /// @notice total vcon deposited per venue
+    mapping(address => uint256) public venueTotalShares;
+
+    /// @notice map an underlying token to the corresponding holding deposit
+    mapping(address => address) public underlyingTokenToHoldingDeposit;
+
+    /// no balance checks when unstaking
+
+    /// ---------- Per Venue User Profit Tracking ----------
+
+    /// @dev convention for all double nested address mappings is key (venue -> user) -> value
+
+    /// @notice record how much VCON a user deposited in a given venue
+    mapping(address => mapping(address => uint256)) public venueUserShares;
+
+    /// @param _core reference to core
+    /// @param _pcvRouter reference to pcvRouter
+    constructor(address _core, address _pcvRouter) CoreRefV2(_core) {
+        pcvRouter = _pcvRouter;
+    }
+
+    /// @notice update the VCON share price and the last recorded profit for a given venue
+    /// @param venue address to accrue
+    function accrueVcon(address venue) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+
+        require(oracle.isVenue(venue), "MarketGovernance: invalid destination");
+
+        _accrue(venue);
+    }
+
+    /// ---------- Permissionless User PCV Allocation Methods ----------
+
+    /// @notice a user can get slashed up to their full VCON stake for entering
+    /// a venue that takes a loss.
+    /// @param amountVcon to stake on destination
+    /// @param destination address to accrue rewards to, and send funds to
+    function stake(
+        uint256 amountVcon,
+        address destination
+    ) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+        require(
+            oracle.isVenue(destination),
+            "MarketGovernance: invalid destination"
+        );
+
+        _accrue(destination); /// update share price in the destination so the user gets in at the current share price
+
+        /// vconToShares will always return correctly as accrue will set venueLastRecordedVconSharePrice
+        /// to the correct share price from 0 if uninitialized
+        uint256 userShareAmount = vconToShares(destination, amountVcon);
+
+        /// user updates
+        venueUserShares[destination][msg.sender] += userShareAmount;
+
+        /// venue updates
+        venueTotalShares[destination] += userShareAmount;
+
+        /// check and an interaction with a trusted contract
+        vcon().safeTransferFrom(msg.sender, address(this), amountVcon); /// transfer VCON in
+
+        emit Staked(destination, msg.sender, amountVcon);
```

Comment:
shouldn't this be `VconStaked` ? the events seem redundant so maybe remove `VconStaked`